protocol/communities/persistence.go

```diff
@@ -2092,3 +2099,101 @@ func (p *Persistence) GetCommunityRequestsToJoinRevealedAddresses(communityID []
 
 	return accounts, nil
 }
+
+func (p *Persistence) GetEncryptionKeyRequests(communityID []byte, channelIDs map[string]struct{}) (map[string]*EncryptionKeysRequestRecord, error) {
+	result := map[string]*EncryptionKeysRequestRecord{}
+
+	query := "SELECT channel_id, requested_at, requested_count FROM community_encryption_keys_requests WHERE community_id = ? AND channel_id IN (?" + strings.Repeat(",?", len(channelIDs)-1) + ")"
+
+	args := make([]interface{}, 0, len(channelIDs)+1)
+	args = append(args, communityID)
+	for channelID := range channelIDs {
+		args = append(args, channelID)
+	}
+
+	rows, err := p.db.Query(query, args...)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var channelID string
+		var requestedAt int64
+		var requestedCount uint
+		err := rows.Scan(&channelID, &requestedAt, &requestedCount)
+		if err != nil {
+			return nil, err
+		}
+		result[channelID] = &EncryptionKeysRequestRecord{
+			communityID:    communityID,
+			channelID:      channelID,
+			requestedAt:    requestedAt,
+			requestedCount: requestedCount,
+		}
+	}
+
+	err = rows.Err()
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (p *Persistence) UpdateEncryptionKeyRequests(communityID types.HexBytes, channelIDs []string, requestedAt int64) error {
+	tx, err := p.db.Begin()
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+
+	if len(channelIDs) == 0 {
+		deleteQuery := "DELETE FROM community_encryption_keys_requests WHERE community_id = ?"
```

Comment:
This use is not super intuitive. Should we add a `delete` function or rename this one to `UpdateOrDelete`?

Reply:
Or maybe just renaming to `UpdateEncryptionKeyRequestsForCommunity` at least makes it clear that we are setting the entries for that community

Reply:
Good point, renamed to `UpdateAndPruneEncryptionKeyRequests`.